### DISCUSSION TITLE
macOS support

### DISCUSSION THREAD

**Discussion by gaurav202015** [EnviralDesign/GeoPix#25] [Q&A]
Dear developer : Can install Geopix 2 on Mac OS X if yes then is it possible to install on M1 machine as well

**Comment by EnviralDesign** [comment]
Hi @gaurav202015 currently OSX support is not implemented. it is planned to be in eventually, but development for one OS is an intensive task, so I am planning to hold off on OSX until the feature set and bugs in 2.x settles and stabilizes.

If anyone else reading this is very interested in an OSX compatible version please chime in! this sort of feedback is invaluable and does tip the scales in what I prioritize and how soon etc.

**Reply by FIDLighting** [reply]
I am also interested but I am curious, if you are running geopix on top of TD can I open TD on a mac and compile it manually? I have no experience with this but I'm down to try it its possible

**Reply by wildeyes** [reply]
@EnviralDesign hey, what would porting GeoPix might entail?
I'm seeing the batch file installs TouchDesigner, and the repo uses python and tkinter

**Reply by steffano** [reply]
👀

**Reply by gundullla** [reply]
very much interested in macOS version

**Comment by zephirl** [comment]
Hey! I would be very interested in a Mac version as well!

**Reply by EnviralDesign** [reply]
awesome appreciate the input!

**Comment by FedericoMurgia** [comment]
Hi I would be very interested in using GeoPix on mac, isn't it possible yet?

**Comment by jessesep** [comment]
@EnviralDesign I'm in heavy need of a MacOS version :')

**Comment by interim17** [comment]
Would really really love this on Mac, what an amazing project

**Comment by philspitlerSF** [comment]
I'd also love an OSX version.

**Comment by batmundo** [comment]
+1 for a Mac version!

**Comment by mylesdebastion** [comment]
Try the vulcan branch, the Geopix toe file should open in the Mac version
of Touchdesigner.

**Reply by billy-jwdd** [reply]
can you explain better? im interested in using this extension

**Reply by jeancongkha** [reply]
I downloaded and tested the file "GeoPix_2.0.toe". It ran through TouchDesigner but reported an error (picture attached). Looking forward to the MacOS version.
<img width="959" alt="Screenshot 2024-01-14 at 14 24 48" src="https://github.com/EnviralDesign/GeoPix/assets/26606392/58239f14-738a-4c43-b1b3-7fde28217a62">
<img width="1316" alt="Screenshot 2024-01-14 at 14 25 47" src="https://github.com/EnviralDesign/GeoPix/assets/26606392/97d675bf-312a-4ac1-8867-8ae6fb3fb5fa">

**Reply by ewoudwijma** [reply]
> Try the vulcan branch, the Geopix toe file should open in the Mac version of Touchdesigner.
> […](#)
> Message ID: <EnviralDesign/GeoPix/repo-discussions/25/comments/7353727@ github.com>

Steps done:
- cloned the geopix repo: https://github.com/EnviralDesign/GeoPix
- find the right version of touchdesigner (Vulkan) in START_GEOPIX.bat
```
SET td_year=2021
SET td_version=16410
SET geopix_executable=GeoPix_2.0.toe
```
- go to https://derivative.ca/download/archive#tab-1-18 to download and install this version on macOS (M2)
- find GeoPix_2.0.toe in the repo clone and open it with TouchDesigner
- GeoPix screen opens, select File Load:

![image](https://github.com/user-attachments/assets/9724f82e-0acc-4fb5-90bb-c980c5e49659)

- This screen appears to expect windows file systems, not possible to navigate through it
- entering /Users/<users>/Downloads works
- but selecting / double clicking a .gp file in that folder is not working

So here I got stuck, any workarounds / tricks (emulate windows FS??) ?

@EnviralDesign , is supporting MacOS FS planned?

**Reply by user06754** [reply]
+1 for OSX Version @EnviralDesign

**Reply by gundullla** [reply]
> Try the vulcan branch, the Geopix toe file should open in the Mac version of Touchdesigner.
> […](#)
> Message ID: <EnviralDesign/GeoPix/repo-discussions/25/comments/7353727@ github.com>

did you get it to run without any bugs?  if so how and on which system? it did not work for me.
